sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/src/main/java/com/microsoft/opentelemetry/exporter/azuremonitor/AzureMonitorExporterBuilder.java

```diff
@@ -134,15 +136,31 @@ public AzureMonitorExporterBuilder configuration(Configuration configuration) {
     }
 
     /**
-     * Sets the instrumentation key to use for exporting telemetry events to Azure Monitor.
-     * @param instrumentationKey The instrumentation key of the Azure Monitor resource.
+     * The connection string to use for exporting telemetry events to Azure Monitor.
+     * @param connectionString The connection string for the Azure Monitor resource.
      * @return The updated {@link AzureMonitorExporterBuilder} object.
+     * @throws NullPointerException If the connection string is {@code null}.
+     * @throws IllegalArgumentException If the connection string is invalid.
      */
-    public AzureMonitorExporterBuilder instrumentationKey(String instrumentationKey) {
-        this.instrumentationKey = instrumentationKey;
+    public AzureMonitorExporterBuilder connectionString(String connectionString) {
+        this.instrumentationKey = extractInstrumentationKey(connectionString);
```

Comment:
I think ideally we would also extract `IngestionEndpoint` from the connection string and not expose endpoint() directly